File: Python3/1921_eliminate_maximum_number_of_monsters.py
# https://leetcode.cn/problems/eliminate-maximum-number-of-monsters/

class Solution:
    def eliminateMaximum(self, dist, speed):
        """
        Inputs:
            dist, speed: list[int]
        Outputs:
            res: int
        """
        n = len(dist)
        t = [(dist[i] + speed[i] - 1) // speed[i] for i in range(n)]
        t.sort()
        res = 0
        for i in range(n):
            if t[i] - i > 0:
                res += 1
            else:
                break
        return res

File: Cpp/1921_eliminate_maximum_number_of_monsters.cpp
// https://leetcode.cn/problems/eliminate-maximum-number-of-monsters/

class Solution {
public:
    int eliminateMaximum(vector<int>& dist, vector<int>& speed) {
        int n = dist.size();
        vector<int> t(n);
        for (int i = 0; i < n; ++i) {
            t[i] = (dist[i] + speed[i] - 1) / speed[i];
        }
        sort(t.begin(), t.end());
        int res = 0;
        for (int i = 0; i < n; ++i) {
            if (t[i] - i > 0) {
                res += 1;
            }
            else {
                break;
            }
        }
        return res;
    }
};